### PR TITLE
Wi-SUN api documentation changes

### PR DESCRIPTION
**Pull request by KariHaapalehto**

### CODE REVIEW

File: docs/api/networkinterfaces/MeshInterface.md [hunk 1]
```diff
@@ -4,14 +4,15 @@
...
 - Thread, following the specification from Thread Group.
 
-Nanostack is the networking stack that provides both of these protocols. For more information on the stack internals, please refer to the [6LoWPAN mesh technology](../reference/mesh-tech.html) section. Application developers use Nanostack through the Mbed Mesh API.
+Nanostack is the networking stack that provides these protocols. For more information on the stack internals, please refer to the [mesh technology](../reference/mesh-tech.html) section. Application developers use Nanostack through the Mbed Mesh API.
 
-The application can use the `LoWPANNDInterface` or `ThreadInterface` object for connecting to the mesh network. When successfully connected, the application can use the Mbed [C++ socket APIs](network-socket.html) to create a socket to start communication with a remote peer.
+The application can use the `LoWPANNDInterface`, WisunInterface` or `ThreadInterface` object for connecting to the mesh network. When successfully connected, the application can use the Mbed [C++ socket APIs](network-socket.html) to create a socket to start communication with a remote peer.
```
Comment (SeppoTakalo) [inline docs/api/networkinterfaces/MeshInterface.md]: WisunInterface missing the first `

File: docs/reference/configuration/mesh.md [hunk 2]
```diff
@@ -276,3 +278,69 @@ Name: mbed-mesh-api.6lowpan-nd-security-mode
     Macro name: MBED_CONF_MBED_MESH_API_6LOWPAN_ND_SECURITY_MODE
     Value: NONE (set by library:mbed-mesh-api)
 ```
+
+#### Wi-SUN related configuration parameters
```
Comment (SeppoTakalo) [inline docs/reference/configuration/mesh.md:296]: Configuration should be moved to their own page.
But I don't find the "Configuration" page anymore from os.mbed.com/docs.. it has been very well hidden.

File: docs/api/networkinterfaces/MeshInterface.md (hunk 1 above)
Comment (KariHaapalehto) [reply docs/api/networkinterfaces/MeshInterface.md]: corrected

File: docs/reference/configuration/mesh.md (hunk 2 above)
Comment (KariHaapalehto) [reply docs/reference/configuration/mesh.md:296]: But this is the place, were the thread and 6lowpan configurations are

Comment (SeppoTakalo) [reply docs/reference/configuration/mesh.md:296]: Maybe it deserves a separate PR.

**Review by SeppoTakalo — APPROVED** [review]

**Comment by SeppoTakalo** [comment]
@mikter  Please review, or delegate someone from your team.

**Comment by AnotherButler** [comment]
Tagging @kegilbert to make sure this works with his config script

**Comment by AnotherButler** [comment]
Note to self: No code dependency because code already merged: https://github.com/ARMmbed/mbed-os/pull/9838

**Comment by kegilbert** [comment]
There's a few sections as is that will pose issues with the configuration parameter auto-update script. 

The tool works by collecting all markdown code blocks and attempting to grab the prefix from the block. That prefix is then used to collect the output of `mbed compile --config -v --prefix <PREFIX>` and replace the old configuration parameter list. This allows us to keep the list up to date each patch release with minimal effort.

The downsides to this script are mainly:
1. [Minor] Can't have multiple prefixes in a single code block. For example in configuration/mesh.md,
```
.....
Name: nanostack-hal.nvm_cfstore
    Description: Use cfstore as a NVM storage. Else RAM simulation will be used
    Defined by: library:nanostack-hal
    No value set
Name: nanostack.configuration
    Description: Build time configuration. Refer to Handbook for valid values. Default: full stack
    Defined by: library:nanostack
    Macro name: MBED_CONF_NANOSTACK_CONFIGURATION
    Value: nanostack_full (set by library:nanostack)
```
would be a problem since the `nanostack` parameter would be dropped (only `nanostack-hal` would be run). With some more work on the tool this could be changed if needed. The current work around would just be to move the prefix to its own code block segment.

2. [Major] You cannot split up a lengthy list such as `mbed-mesh-api` into more logical sections as it is done now in configuration/mesh.md as each section will just be replaced by the entirety of the output from `mbed compile --config -v --prefix <PREFIX>`. 

The workaround for that was to combine all of the separated sections into one segment, loosing some depth and clarity in our ability to highlight or group together sections of configuration parameters, but guaranteeing an always up to date list of parameters.

**Comment by AnotherButler** [comment]
@KariHaapalehto Can you please make the requested changes, so our config update script works with this doc?

**Comment by KariHaapalehto** [comment]
@SeppoTakalo , could you please comment about [2].
I don't like, that we should combine all mesh-api configurations to one big table just because of the tool. It doesn't make our document any better. Mesh-api contains some generic configuration values and 3 separate interface related configurations. In my opinion they are now well grouped and should be kept separate. Should we just run the tool to update parameter lists and keep them separate?

**Comment by AnotherButler** [comment]
I'd like to keep all of our pages uniform, up to date and reflective of the code. We do have pages with separations (such as [this one](https://os.mbed.com/docs/mbed-os/development/reference/configuration-rtos.html)), but they're still grouped by code prefix. In this example, we've put the ones that start with "rtos" in one group and the ones that start with "events" in another.

Because the mesh parameters all start with the same prefix, it makes sense for them to belong together. It's painful to remember to manually change one page when we run a script on the others. Also, we eventually hope to put the script into CI, which would make manual changes difficult.

**Comment by mikter** [comment]
@artokin could you check these.

Also there was changes in the Wi-SUN/Thread documentations and I think here is a link to those. are these links still functional?

**Comment by SeppoTakalo** [comment]
I object against combining the values into one list.
This affects the readability too much. Makes it almost unreadable.

We should not choose between readability and maintainability.

**Comment by KariHaapalehto** [comment]
I just pushed update to this pr. I did run the tool and config blocks are updated from the tool output. Nanostack configuration is now splitted to separate blocks based on prefix. Mbed-mesh-api is still splitted to logical blocks, since combining them to one big block would affect to readability.

File: docs/reference/configuration/mesh.md [hunk 3]
```diff
@@ -1,16 +1,16 @@
...
 
-This guide is divided in sections, and generic Nanostack configurations are first, followed by those for Thread and then 6LoWPAN-ND.
+This guide is divided in sections, and generic Nanostack configurations are first, followed by those for Thread, 6LoWPAN-ND and then Wi-SUN.
 
 For understanding the technologies and APIs, please refer to following sections before this one:
 
 - [Network connectivity in Mbed OS](../reference/networking.html) technology page.
-- [6LoWPAN Mesh technology](../reference/mesh-tech.html) page.
-- [6LoWPAN Mesh class reference](../apis/mesh-api.html) user API.
+- [Mesh technology](../reference/mesh-tech.html) page.
```
Comment (artokin) [inline docs/reference/configuration/mesh.md]: This has been changed, now 6LoWPAN_ND, Wi-SUN and Thread are having separate page

File: docs/reference/configuration/mesh.md [hunk 4]
```diff
@@ -53,11 +53,12 @@ Option name | Features supported | Estimated binary size of Nanostack
 
 If you want to optimize the flash usage, you need to configure Nanostack. The configuration to choose depends mostly on the preferred use case.
 
-See the [6LoWPAN technology overview](mesh-tech.html) for the definition of star and mesh networks. These same principles apply also to the Thread protocol.
+See the [mesh technology overview](mesh-tech.html) for the definition of star and mesh networks. These same principles apply also to the Wi-SUN and Thread protocols.
```
Comment (artokin) [inline docs/reference/configuration/mesh.md:58]: mesh-tech now contains mainly 6LoWPAN_ND related information

Comment (KariHaapalehto) [reply docs/reference/configuration/mesh.md:58]: star and mesh network definitions are at mesh-tech page, so no need to update this, but I'll update rest of the links

**Comment by AnotherButler** [comment]
@kegilbert has graciously agreed to update his script to make this page more readable.

Once he does, please resolve any merge conflicts.

**Comment by kegilbert** [comment]
PR is up, pending review. Will require a minor modification. On all `mbed_mesh_api` configuration parameter list blocks, please add the following tags following the opening '```' segment on the relevant portions:
`heap`

`thread`

`6lowpan`

`wisun`

Such as: 
___________________

#### Generic Mbed mesh API configuration values

Generic configuration allows you to fine tune Nanostack's heap usage.

'''**heap**
Configuration parameters

Name: mbed-mesh-api.heap-size
    Description: Nanostack's heap size [bytes: 0-65534]
    Defined by: library:mbed-mesh-api
    Macro name: MBED_CONF_MBED_MESH_API_HEAP_SIZE
    Value: 32500 (set by library:mbed-mesh-api)
.....
'''


I can make the changes as well once the merge conflict is resolved if that'd be easier.

**Review by kegilbert — APPROVED** [review]
LGTM from my end, thanks!

**Review by AnotherButler — APPROVED** [review]
Thanks for the PR 👍

**Pull request merged**